server/tests/oauth.test.js

```diff
@@ -0,0 +1,112 @@
+import passport from 'passport';
+import * as chai from 'chai';
+import sinonChai from 'sinon-chai';
+import sinon from 'sinon';
+import request from 'supertest';
+import { googleProfile, facebookProfile } from './mochData/socialMock';
+import data from './mochData/socialMockData';
+import VerifyUser from '../helpers/verifyUser';
+import CheckUser from '../middlewares/checkUser';
+
+import app from '../index';
+
+chai.use(sinonChai);
+chai.should();
+const { expect } = chai;
+
+
+describe('Social authentication tests', () => {
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should return user data when it get the profile from the social account', async () => {
+    const accessToken = 'zzzz-zzzz-zzzz';
+    const refreshToken = 'zzzzzzz';
+    const callBack = sinon.spy();
+    await VerifyUser.inSignSocial(accessToken, refreshToken, googleProfile, callBack);
+    expect(callBack.withArgs(null, googleProfile));
+  });
+});
+
+describe('Social authentication tests', () => {
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should return user data when it get the profile from the social account', async () => {
+    const accessToken = 'zzzz-zzzz-zzzz';
+    const refreshToken = 'zzzzzzz';
+    const callBack = sinon.spy();
+    await VerifyUser.inSignSocial(accessToken, refreshToken, facebookProfile, callBack);
+    expect(callBack.withArgs(null, facebookProfile));
+  });
+});
+
+describe('User sign in successfully with google', () => {
+  before(async () => {
+    sinon.stub(passport, 'authenticate').callsFake((strategy, options, callback) => {
+      callback(null, data[3], null);
+      return () => { };
+    });
+  });
+  after(() => {
+    passport.authenticate.restore();
+  });
+  it('should sign in user', (done) => {
```

Comment:
Update the description to **should successfully sign in the user with Google**

Reply:
Feedback Implemented!